File: InclusiveConnectBackend/src/main/java/com/inclusiveconnect/inclusiveconnectbackend/service/JobApplicationService.java
package com.inclusiveconnect.inclusiveconnectbackend.service;

import com.inclusiveconnect.inclusiveconnectbackend.dto.request.ApplyJobRequest;
import com.inclusiveconnect.inclusiveconnectbackend.dto.request.UpdateApplicationStatusRequest;
import com.inclusiveconnect.inclusiveconnectbackend.dto.response.JobApplicationResponse;

import java.util.List;

public interface JobApplicationService {
    JobApplicationResponse applyToJob(Long candidateUserId, Long jobId, ApplyJobRequest request);
    List<JobApplicationResponse> getMyApplications(Long candidateUserId);
    List<JobApplicationResponse> getApplicationsForJob(Long employerUserId, Long jobId);
    JobApplicationResponse updateApplicationStatus(Long employerUserId, Long applicationId, UpdateApplicationStatusRequest request);
}

File: InclusiveConnectFrontend/src/app/core/services/job-application.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../models/api-response.model';
import { ApplicationStatus, JobApplication } from '../models/job-application.model';

@Injectable({ providedIn: 'root' })
export class JobApplicationService {
  private http = inject(HttpClient);
  private baseUrl = environment.apiUrl;

  applyToJob(jobId: number, coverLetter?: string): Observable<ApiResponse<JobApplication>> {
    return this.http.post<ApiResponse<JobApplication>>(`${this.baseUrl}/jobs/${jobId}/apply`, { coverLetter });
  }

  getMyApplications(): Observable<ApiResponse<JobApplication[]>> {
    return this.http.get<ApiResponse<JobApplication[]>>(`${this.baseUrl}/applications/me`);
  }

  getApplicationsForJob(jobId: number): Observable<ApiResponse<JobApplication[]>> {
    return this.http.get<ApiResponse<JobApplication[]>>(`${this.baseUrl}/jobs/${jobId}/applications`);
  }

  updateApplicationStatus(applicationId: number, status: ApplicationStatus): Observable<ApiResponse<JobApplication>> {
    return this.http.patch<ApiResponse<JobApplication>>(`${this.baseUrl}/applications/${applicationId}/status`, { status });
  }
}